homeassistant/components/cast/__init__.py

```diff
@@ -1,20 +1,42 @@
 """Component to embed Google Cast."""
+import logging
+
+import voluptuous as vol
+
 from homeassistant import config_entries
+from homeassistant.helpers import config_validation as cv
 
 from . import home_assistant_cast
 from .const import DOMAIN
+from .media_player import ENTITY_SCHEMA
+
+# Deprecated from 2021.4, remove in 2021.6
+CONFIG_SCHEMA = cv.deprecated(DOMAIN)
+
+_LOGGER = logging.getLogger(__name__)
 
 
 async def async_setup(hass, config):
     """Set up the Cast component."""
     conf = config.get(DOMAIN)
 
-    hass.data[DOMAIN] = conf or {}
-
     if conf is not None:
+        media_player_config_validated = []
+        media_player_config = conf.get("media_player", {})
+        if not isinstance(media_player_config, list):
+            media_player_config = [media_player_config]
+        for cfg in media_player_config:
+            try:
+                cfg = ENTITY_SCHEMA(cfg)
+                media_player_config_validated.append(cfg)
+            except vol.Error as ex:
+                _LOGGER.warning("Invalid config '%s': %s", cfg, ex)
```

Comment:
Do we accept a partial valid config to be more backwards compatible? The alternative would be to extend the config schema instead which would fail the whole config on invalid result.

Reply:
Yes, the intention is to accept a partial valid config by simply skipping invalid media players instead of rejecting the entire config.